src/vulkan/nvofapi_vulkan_instance.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: MIT
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#include "util/util_log.h"
+#include "util/util_string.h"
+
+#include "../inc/nvofapi/nvOpticalFlowVulkan.h"
+
+#include "vulkan/nvofapi_vulkan_instance.h"
+
+namespace nvofapi {
+
+    bool NvOFInstanceVk::Initialize() {
+        // For VK we cannot load winevulkan directly or we may break handle
+        // opacity.
+        m_library = LoadLibraryA("vulkan-1.dll");
```

Comment:
One more thing about this: I just remembered that Streamline also exports its own VK funcs from `sl.interposer.dll`. This sounds like another possible problem but I have no idea how to detect if some game is using Streamline exports or if it's calling Vulkan loader directly.

Reply:
That's used for Streamline's "implicit hooking". Your intuition that we should be loading `vulkan-1.dll` is correct because that's what other middleware would be doing (presuming it is not passed `vkGetInstanceProcAddr()` directly by the application).